pkg/snapshot/snapshot.go

```diff
@@ -81,16 +94,19 @@ func (s *Snapshotter) snapshotFiles(f io.Writer, files []string) (bool, error) {
 	}
 	logrus.Infof("Taking snapshot of files %v...", files)
 	snapshottedFiles := make(map[string]bool)
+	n := len(files)
 	for _, file := range files {
 		parentDirs := util.ParentDirectories(file)
+		// If we do end up snapshotting the dirs, we need to snapshot them before
+		// we snapshot their contents
 		files = append(parentDirs, files...)
```

Comment:
@priyawadhwa can you confirm if the comment i added is correct? I was trying to understand why we were actually appending to `parentDirs` instead of to `files`

Reply:
ah kk! if we're going to have 2 separate lists it wont matter too much anyway